subsys/nrf_security/Kconfig.legacy

```diff
@@ -76,81 +76,82 @@ config MBEDTLS_AES_DECRYPT_ALT
 
 config MBEDTLS_AES_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_CIPHER_DRIVER
 	default y
 
 config MBEDTLS_CMAC_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_CMAC
 	default y
 
 config MBEDTLS_CCM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_AES_CCM
 	default y
 
 config MBEDTLS_GCM_ALT
 	bool
 	depends on CC312_BACKEND || \
-		   (PSA_CRYPTO_DRIVER_CC3XX && CRYPTOCELL_CC312_USABLE)
+		   (PSA_NEED_CC3XX_AES_GCM && CRYPTOCELL_CC312_USABLE)
 	default y
 
 config MBEDTLS_CHACHA20_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_CHACHA20 || \
+		   CC3XX_BACKEND || PSA_NEED_CC3XX_CHACHA20
 	default y
 
 config MBEDTLS_POLY1305_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_CHACHA20_POLY1305 || \
+		   CC3XX_BACKEND || PSA_NEED_CC3XX_CHACHA20_POLY1305
 	default y
 
 config MBEDTLS_CHACHAPOLY_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_CHACHA20_POLY1305
 	default y
 	depends on MBEDTLS_CHACHA20_ALT
 
 config MBEDTLS_DHM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_RSA_DRIVER
 	default y
 
 config MBEDTLS_ECP_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_ECDSA_DRIVER || \
+		   PSA_NEED_CC3XX_ECDH_DRIVER
 	default y
 
 config MBEDTLS_ECDH_GEN_PUBLIC_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_ECDH_DRIVER || \
+		   CC3XX_BACKEND || PSA_NEED_CC3XX_ECDH_DRIVER
 	default y
 
 config MBEDTLS_ECDH_COMPUTE_SHARED_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_ECDH_DRIVER || \
+		   CC3XX_BACKEND || PSA_NEED_CC3XX_ECDH_DRIVER
 	default y
 
 config MBEDTLS_ECDSA_GENKEY_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_ECDSA_DRIVER || \
```

Comment:
For ECDSA I think that what you need is:
PSA_NEED_CC3XX_KEY_PAIR_SECP || PSA_NEED_CC3XX_KEY_PAIR_SECK1
|| PSA_NEED_OBERON_KEY_PAIR_DRIVER

These are relevant to the key generation. So you can just remove the ECDSA_DRIVER, as it is right now it doesn't provide key generation anyway, it only provides the operations I think.